src/models/discriminant_analysis.jl

```diff
@@ -3,130 +3,123 @@
+        model.cov_b != MS.SimpleCovariance()
+  if nc <= 1 && lone_class_unsupported
+      throw(ERR_LONE_TARGET_CLASS)
+  end
+
+  # Check to make sure we have more samples than classes.
+  # This is to prevent Sw from being the zero matrix.
+  if n <= nc
+      throw(
+          ArgumentError(
+              "The number of training samples `n` has"*
+              " to be greater than number of unique classes `nc`."
+          )
+      )
+  end
+  # Check output dimension default is min(p, nc-1)
+  def_outdim = min(p, nc - 1)
+  # If unset (0) use the default; otherwise try to use the provided one
+  outdim = ifelse(model.outdim == 0, def_outdim, model.outdim)
+  # Check if the given one is sensible
+  if outdim > def_outdim
+      throw(
+          ArgumentError(
+              "`outdim` must not be larger than `min(p, nc-1)`"*
+              "where `p` is the number of features in `X` and"*
+              "`nc` is the number of unique classes in the target vector."
+          )
+      )
+  end
+  return Xm_t, y, classes_seen, pool, p, n, nc, outdim
+end
+
+function MMI.fit(model::LDA, ::Int, X, y)
+  Xm_t, _ , classes_seen, pool, _, _, nc, outdim =
+      _check_lda_data(model, X, y)
+  core_res = MS.fit(
+      MS.MulticlassLDA, Xm_t, y;
+      method=model.method,
+      outdim,
+      regcoef=model.regcoef,
+      covestimator_within=model.cov_w,
+      covestimator_between=model.cov_b
+  )
+  cache = nothing
+  report = (
+      indim = MS.size(core_res)[1],
+      outdim=MS.size(core_res)[2],
+      mean=MS.mean(core_res),
+      nclasses=nc,
+      class_means=MS.classmeans(core_res),
+      class_weights=MS.classweights(core_res),
+      Sb=MS.betweenclass_scatter(core_res),
+      Sw=MS.withclass_scatter(core_res)
+  )
+  fitresult = (core_res, classes_seen, pool)
+  return fitresult, cache, report
 end
 
-function MMI.fitted_params(::LDA, (core_res, classes_seen))
-    return (
-        projected_class_means=MS.classmeans(core_res),
-        projection_matrix=MS.projection(core_res)
-    )
+function MMI.fitted_params(::LDA, (core_res, classes_seen, _))
+  return (
+      classes = classes_seen,
+      projection_matrix=MS.projection(core_res)
+  )
 end
 
-function MMI.predict(m::LDA, (core_res, classes_seen), Xnew)
-    # projection of Xnew, XWt is nt x o  where o = number of out dims
-    # nt = number ot test samples
-    XWt = MMI.matrix(Xnew) * core_res.proj
-    # centroids in the transformed space, nc x o
-    centroids = transpose(core_res.pmeans)
-
-    # compute the distances in the transformed space between pairs of rows
-    # the probability matrix Pr is `n x nc` and normalised accross rows
-    Pr = pairwise(m.dist, XWt, centroids, dims=1)
-    Pr .*= -1
-    # apply a softmax transformation
-    softmax!(Pr)
-    return MMI.UnivariateFinite(classes_seen, Pr)
+function MMI.predict(m::LDA, (core_res, classes_seen, pool), Xnew)
+  # projection of Xnew, XWt is nt x o  where o = number of out dims
```

Comment:
4 space indentation  please

Reply:
I'm sorry about the indentation. I don't know how that got changed. I'll fix the indentation.

Reply:
It appears My VSCode indentation was somehow set to 2spaces.